packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -588,33 +598,53 @@ export class ApexDebug extends LoggingDebugSession {
     response: DebugProtocol.SetBreakpointsResponse,
     args: DebugProtocol.SetBreakpointsArguments
   ): Promise<void> {
-    if (args.source && args.source.path && args.lines) {
-      const uri = this.convertClientPathToDebugger(args.source.path);
-      const validBreakpoints = await this.myBreakpointService.reconcileBreakpoints(
-        this.sfdxProject,
-        uri,
-        this.mySessionService.getSessionId(),
-        args.lines.map(line => this.convertClientLineToDebugger(line))
-      );
-      const processedBreakpoints: DebugProtocol.Breakpoint[] = [];
-      validBreakpoints.forEach(validBreakpoint => {
-        processedBreakpoints.push({
-          verified: true,
-          source: args.source,
-          line: this.convertDebuggerLineToClient(validBreakpoint)
-        });
-      });
-      args.lines.forEach(lineArg => {
-        if (!validBreakpoints.has(lineArg)) {
-          processedBreakpoints.push({
-            verified: false,
-            source: args.source,
-            line: this.convertDebuggerLineToClient(lineArg)
+    response = await this.lock.acquire(
+      'breakpoint',
+      async () => {
+        if (args.source && args.source.path && args.lines) {
+          response.body = { breakpoints: [] };
+          const uri = this.convertClientPathToDebugger(args.source.path);
+          const unverifiedBreakpoints: number[] = [];
+          const verifiedBreakpoints = await this.myBreakpointService.reconcileBreakpoints(
+            this.sfdxProject,
+            uri,
+            this.mySessionService.getSessionId(),
+            args.lines.map(line => this.convertClientLineToDebugger(line))
+          );
+          verifiedBreakpoints.forEach(verifiedBreakpoint => {
+            const lineNumber = this.convertDebuggerLineToClient(
+              verifiedBreakpoint
+            );
+            response.body.breakpoints.push({
+              verified: true,
+              source: args.source,
+              line: lineNumber
+            });
+          });
+          args.lines.forEach(lineArg => {
+            if (!verifiedBreakpoints.has(lineArg)) {
+              const lineNumber = this.convertDebuggerLineToClient(lineArg);
+              response.body.breakpoints.push({
+                verified: false,
+                source: args.source,
+                line: lineNumber
+              });
+              unverifiedBreakpoints.push(lineNumber);
+            }
           });
+          this.log(
+            TRACE_CATEGORY_BREAKPOINTS,
+            `setBreakPointsRequest: uri=${uri} args.lines=${args.lines.join(
+              ','
+            )} verified=${Array.from(verifiedBreakpoints).join(
+              ','
+            )} unverified=${unverifiedBreakpoints.join(',')}`
+          );
         }
-      });
-      response.body = { breakpoints: processedBreakpoints };
-    }
+        return Promise.resolve(response);
+      },
+      { timeout: this.breakpointLockTimeout }
```

Comment:
This is either the default or configured in workspace settings.

Reply:
What is the difference between breakpointLockTimeout and DEFAULT_LOCK_TIMEOUT_MS

Reply:
I'll stick with default for now.